src/library_sockfs.js

```diff
@@ -634,7 +634,8 @@ addToLibrary({
         // connect, and lie, saying the data was sent now.
         if (!dest || dest.socket.readyState !== dest.socket.OPEN) {
           // if we're not connected, open a new connection
-          if (sock.type === {{{ cDefs.SOCK_DGRAM }}}) {
+          if (!dest || sock.type === {{{ cDefs.SOCK_DGRAM }}}) {
+            sock.type = {{{ cDefs.SOCK_DGRAM }}};
```

Comment:
It seem lik you could get here with `!dest` and with `sock.type = {{{ cDefs.SOCK_STREAM }}}`.. in which case this line would change the type of the socket, no?

Reply:
Yes, but the check was specific for `sock.type === {{{ cDefs.SOCK_DGRAM }}}` only, that's why I forced it. Maybe it's ok to keep the socket type, then.